fastlane/lib/fastlane/actions/unlock_keychain.rb

```diff
@@ -42,7 +42,12 @@ def self.add_keychain_to_search_list(keychain_path)
       end
 
       def self.replace_keychain_in_search_list(keychain_path)
-        Actions.lane_context[Actions::SharedValues::ORIGINAL_DEFAULT_KEYCHAIN] = Fastlane::Actions.sh("security default-keychain", log: false).strip
+        begin
+          UI.message("Reading existing default keychain")
+          Actions.lane_context[Actions::SharedValues::ORIGINAL_DEFAULT_KEYCHAIN] = Fastlane::Actions.sh("security default-keychain").strip
+        rescue => e
+          raise unless e.message.include?("security: SecKeychainCopyDefault: A default keychain could not be found.")
```

Comment:
Is there another way we could recognize this error besides checking the error message? Maybe the status code?

Reply:
Hi @mollyIV 
That would've been nice, but when I print the exception it just prints this
```
error : Exit status of command 'security default-keychain' was 1 instead of 0.
security: SecKeychainCopyDefault: A default keychain could not be found.
```

I don't know if "1" is a unique error code for this case and therefore I added a check for the error message instead. Maybe you know if we can just check for error code? Also, I don't know how to actually get that error code from the `e` object in `begin-rescue`. 

Here is some documentation on what I believe the `security default-keychain` command calls:
https://developer.apple.com/documentation/security/seckeychaincopydefault(_:)?language=objc

Reply:
I see, thanks for the explanation. Not sure if the status code 1 is the right on, sounds pretty generic. I think checking error description in this case will do.

Reply:
@mollyIV I agree. Thanks a lot for the review!